### PR TITLE
root_generate_dictionary: Improve propagation of INCLUDE_DIRECTORY properties from target and dependencies.

### LINKED ISSUE #12263: root_generate_dictionary does not properly handle "all" include directory properties from its dependencies.
If we have
```
    root_generate_dictionary(HitClassesRootInterfaces
      ${CMAKE_CURRENT_SOURCE_DIR}/demo-geant-integration/HitRootIO.hh
      ${CMAKE_CURRENT_SOURCE_DIR}/demo-geant-integration/SensitiveHit.hh
      NOINSTALL
      MODULE demo-geant-integration
      DEPENDENCIES Celeritas::accel ${Geant4_LIBRARIES} Celeritas::corecel 
      LINKDEF "${CMAKE_CURRENT_SOURCE_DIR}/demo-geant-integration/HitClassesLinkDef.h"
    )
```
`root_generate_dictionary` will call (if they are proper target at that point) `get_property(dep_include_dirs TARGET ${dep} PROPERTY INCLUDE_DIRECTORIES)`

However there is 2 problems.
(a) It ignores [INTERFACE_INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html) and [INTERFACE_SYSTEM_INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_SYSTEM_INCLUDE_DIRECTORIES.html); the former is used by `Geant4` (as it should) to publish its installed directories.

(b) It fails the properties contains cmake generator expressions that does not expand yet.  For example `$<INSTALL_INTERFACE:include>`; it fails badly because it then passed to rootcling a `-I` followed by 'nothing' hence shallowing the next arguments.

### PR DESCRIPTION
**Pull request by pcanal**
This fixes #12263 at least as tested from the Celeritas builds.

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
@phsft-bot build

File: builtins/xrootd/CMakeLists.txt [hunk 1]
```diff
@@ -76,7 +76,9 @@ set(XROOTD_LIBRARY ${XROOTD_PREFIX}/${XROOTD_LIBDIR}/${XROOTD_LIBNAME} CACHE INT
 set(XROOTD_LIBRARIES ${XROOTD_LIBRARIES} CACHE INTERNAL "" FORCE)
 
 add_library(xrootd INTERFACE)
-target_include_directories(xrootd INTERFACE $<BUILD_INTERFACE:${XROOTD_INCLUDE_DIR}>)
+foreach(dir ${XROOT_INCLUDE_DIR})
```
Comment (bellenot) [inline builtins/xrootd/CMakeLists.txt:79]: Is it `${XROOT_INCLUDE_DIR}` or `${XROOTD_INCLUDE_DIR}`?

**Review by bellenot — COMMENTED** [review]
Beside my question, I trust you if it works, even if I'm not sure that makes our build system simpler...

File: builtins/xrootd/CMakeLists.txt (hunk 1 above)
Comment (pcanal) [reply builtins/xrootd/CMakeLists.txt:79]: It is the later :(

Comment (pcanal) [inline builtins/xrootd/CMakeLists.txt:79]:
```suggestion
foreach(dir ${XROOTD_INCLUDE_DIR})
```

**Review by bellenot — APPROVED** [review]
Fine with me if the failures are unrelated...

**Comment by pcanal** [comment]
The mac failure were due to out-of-disk-space issues.

**Pull request merged**